From 2c683ef91cfa757c1c0b03f5d66d0d89b419a4bb Mon Sep 17 00:00:00 2001
From: Jun Wu <jun.wu@arm.com>
Date: Wed, 11 Sep 2024 16:20:40 +0000
Subject: [PATCH] poky/meta: Build gnu-efi by binutils recipe instead of
 HOSTTOOLS

To remove the dependency on host environment, use binutils built
binary objcopy to compile gnu-efi-native recipe.

The issue was found when build gnu-efi-native in an aarch64 host.
If the host has binutils version < 2.38, the gnu-efi-native compile
will fail.

Upstream-Status: Pending

Signed-off-by: Jun Wu <jun.wu@arm.com>
---
 meta/recipes-bsp/gnu-efi/gnu-efi_3.0.17.bb      |  2 ++
 meta/recipes-devtools/binutils/binutils_2.42.bb | 13 -------------
 2 files changed, 2 insertions(+), 13 deletions(-)

diff --git a/meta/recipes-bsp/gnu-efi/gnu-efi_3.0.17.bb b/meta/recipes-bsp/gnu-efi/gnu-efi_3.0.17.bb
index 43b7cc7529..ec27c0b163 100644
--- a/meta/recipes-bsp/gnu-efi/gnu-efi_3.0.17.bb
+++ b/meta/recipes-bsp/gnu-efi/gnu-efi_3.0.17.bb
@@ -24,6 +24,8 @@ SRC_URI[sha256sum] = "7807e903349343a7a142ebb934703a2872235e89688cf586c032b0a108
 COMPATIBLE_HOST = "(x86_64.*|i.86.*|aarch64.*|arm.*|riscv64.*)-linux"
 COMPATIBLE_HOST:armv4 = 'null'
 
+DEPENDS += "binutils-native"
+
 do_configure:linux-gnux32:prepend() {
 	cp ${STAGING_INCDIR}/gnu/stubs-x32.h ${STAGING_INCDIR}/gnu/stubs-64.h
 	cp ${STAGING_INCDIR}/bits/long-double-32.h ${STAGING_INCDIR}/bits/long-double-64.h
diff --git a/meta/recipes-devtools/binutils/binutils_2.42.bb b/meta/recipes-devtools/binutils/binutils_2.42.bb
index 2cce40f1ef..2064f0218a 100644
--- a/meta/recipes-devtools/binutils/binutils_2.42.bb
+++ b/meta/recipes-devtools/binutils/binutils_2.42.bb
@@ -42,19 +42,6 @@ do_install:class-native () {
 	install -d ${D}${includedir}
 	install -m 644 ${S}/include/ansidecl.h ${D}${includedir}
 	install -m 644 ${S}/include/libiberty.h ${D}${includedir}
-
-	# We only want libiberty, libbfd and libopcodes
-	rm -rf ${D}${bindir}
-	rm -rf ${D}${prefix}/${TARGET_SYS}
-	rm -rf ${D}${prefix}/lib/ldscripts
-	rm -rf ${D}${prefix}/share/info
-	rm -rf ${D}${prefix}/share/locale
-	rm -rf ${D}${prefix}/share/man
-	rmdir ${D}${prefix}/share || :
-	rmdir ${D}/${libdir}/gcc-lib || :
-	rmdir ${D}/${libdir}64/gcc-lib || :
-	rmdir ${D}/${libdir} || :
-	rmdir ${D}/${libdir}64 || :
 }
 
 # libctf races with libbfd
-- 
2.25.1

